src/main/resources/fileTemplates/testMeTests/JUnit4 & Powermock.java.ft

```diff
@@ -0,0 +1,94 @@
+#parse("TestMe macros.java")
+#set($hasMocks=$MockitoMockBuilder.hasMocks($TESTED_CLASS))
+#if($PACKAGE_NAME)
+package ${PACKAGE_NAME};
+#end
+
+import org.junit.Assert;
+import org.junit.Test;
+#if($hasMocks)
+import static org.powermock.api.mockito.PowerMockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.junit.Before;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.junit.runner.RunWith;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+#end
+
+#parse("File Header.java")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({${TESTED_CLASS.name}.class})
+@PowerMockIgnore("javax.management.*")
+public class ${CLASS_NAME} {
+    #renderMockedFields($TESTED_CLASS.fields,$hasMocks)
+    #renderPowerMockTestSubjectInit($TESTED_CLASS,$TestSubjectUtils.hasTestableInstanceMethod($TESTED_CLASS.methods),$hasMocks)
+
+    ## all methods under powermock framework can be tested
+    #foreach($method in $TESTED_CLASS.methods)
+        #if(!$method.inherited)
+            @Test
+            public void #renderTestMethodName($method.name)() throws Exception {
+                #if($hasMocks && $MockitoMockBuilder.shouldStub($method, $TESTED_CLASS.fields))
+                    #renderMockStubs($method, $TESTED_CLASS.fields)
+                #end
+                #renderSelfCallMethodMockStubs($method, $TESTED_CLASS)
+                #if($method.isPrivate())
+                    #renderPrivateMethodCall($method, $TESTED_CLASS.name)
+                #else
+                    #renderMethodCall($method, $TESTED_CLASS.name)
+                #end
+                #if($hasMocks && $MockitoMockBuilder.shouldVerify($method,$TESTED_CLASS.fields))
+                    #renderMockVerifies($method,$TESTED_CLASS.fields)
+                #end
+                #if($method.hasReturn())
+                    Assert.#renderJUnitAssert($method)
+                #end
+            }
+      #end
+    #end
+}
+
+################## Macros #####################
+#macro(renderPowerMockTestSubjectInit $testedClass $hasTestableInstanceMethod $hasMocks)
+    #if($hasMocks)
+        @InjectMocks
```

Comment:
don't think it makes sense to call @InjectMock and initialize with a constructor at the same time. @InjectMock is supposed to initialize the tested class before injecting mocks.